tests/fixtures/optimizer/rewrite_subqueries.sql

```diff
@@ -0,0 +1,14 @@
+SELECT * FROM x AS x WHERE (SELECT y.a FROM y AS y WHERE x.a = y.a) = 1;
+SELECT * FROM x AS x JOIN (SELECT y.a FROM y AS y WHERE TRUE GROUP BY y.a) AS "_d_0" ON _d_0.a = x.a AND ("_d_0".a) = 1 WHERE TRUE;
```

Comment:
Should we be quoting `_d_0` in the condition too?

Reply:
i thought about that but doesn’t it happen in a later optimization step which quotes everything?

Reply:
Ah yeah sure,